### PR TITLE
Extract ucfg paths from list of issues

### PR DESCRIPTION
**Pull request by victor-diez-sonarsource**
Need to merge the ignore of ucfg "issues" to have sonar-security pass QA testing

https://github.com/SonarSource/sonar-security/pull/3351

### CODE REVIEW

File: eslint-bridge/jest.config.js [hunk 1]
```diff
@@ -15,4 +15,5 @@ module.exports = {
     '^.+\\.ts$': 'ts-jest',
   },
   testMatch: ['<rootDir>/tests/**/*.test.ts'],
+  testTimeout: 20000,
```
Comment (vilchik-elena) [inline eslint-bridge/jest.config.js:18]: why?

File: eslint-bridge/src/linting/eslint/linter/issues/transform.ts [hunk 2]
```diff
@@ -22,29 +22,79 @@ import { Linter, Rule, SourceCode } from 'eslint';
...
+ *
+ * ESLint API returns what it calls messages as results of linting a file.
+ * A linting result in the context of the analyzer includes more than that
+ * as it needs not only to transform ESLint messages into SonarQube issues
+ * as well as analysis data about the analyzed source code, namely symbol
+ * highlighting and cognitive complexity.
+ *
+ * @param issues the issues found in the code
+ * @param highlightedSymbols the symbol highlighting of the code
+ * @param cognitiveComplexity the cognitive complexity of the code
```
Comment (vilchik-elena) [inline eslint-bridge/src/linting/eslint/linter/issues/transform.ts:39]: should we add `ucfgPaths` here and in the textual description?

File: eslint-bridge/src/linting/eslint/linter/issues/transform.ts [hunk 3]
```diff
@@ -22,29 +22,79 @@ import { Linter, Rule, SourceCode } from 'eslint';
...
 
 /**
  * Transforms ESLint messages into SonarQube issues
  *
+ * The result of linting a source code requires post-linting transformations
+ * to return SonarQube issues. These transformations include extracting ucfg
+ * paths, decoding issues with secondary locations as well as converting
+ * quick fixes.
+ *
+ * Besides issues, a few metrics are computing during linting in the form of
```
Comment (vilchik-elena) [inline eslint-bridge/src/linting/eslint/linter/issues/transform.ts:56]: computing -> computed

File: eslint-bridge/tests/routing/routing.test.ts [hunk 4]
```diff
@@ -25,7 +25,7 @@ import path from 'path';
 import { start } from 'server';
 import { createProgram } from 'services/program';
 import { promisify } from 'util';
-import { request, toUnixPath } from '../tools/helpers';
+import { request, toUnixPath } from '../tools';
```
Comment (vilchik-elena) [inline eslint-bridge/tests/routing/routing.test.ts:28]: is this related to the PR?

File: eslint-bridge/tests/linting/eslint/linter/issues/transform.test.ts [hunk 5]
```diff
@@ -36,7 +36,9 @@ describe('transformMessages', () => {
     const linter = new Linter();
     const messages = linter.verify(sourceCode, config);
 
-    const [issue] = transformMessages(messages, { sourceCode, rules: linter.getRules() });
+    const {
+      issues: [issue],
+    } = transformMessages(messages, { sourceCode, rules: linter.getRules() });
```
Comment (vilchik-elena) [inline eslint-bridge/tests/linting/eslint/linter/issues/transform.test.ts:41]: that's definitely minor but this destructuring looks bulky, I would use simple `.issues` on left-hand-side (`const [issue] = transform().issues`)

File: eslint-bridge/tests/linting/eslint/linter/issues/transform.test.ts [hunk 6]
```diff
@@ -133,4 +141,30 @@ describe('transformMessages', () => {
...
+    const tsConfigs = [];
+    const sourceCode = parseTypeScriptSourceFile(filePath, tsConfigs) as SourceCode;
+
+    const ruleId = 'no-duplicate-in-composite';
+    const config = { rules: { [ruleId]: 'error' } } as any;
+
+    const linter = new Linter();
+    linter.defineRule(ruleId, noDuplicateInComposite);
+
+    const messages = linter.verify(sourceCode, config);
```
Comment (vilchik-elena) [inline eslint-bridge/tests/linting/eslint/linter/issues/transform.test.ts:156]: I don't see much point in actual execution of linter if we can simply create messages manually.

**Review by vilchik-elena — APPROVED** [review]
LGTM, some cosmetic improvements could be made

File: eslint-bridge/tests/routing/routing.test.ts (hunk 4 above)
Comment (victor-diez-sonarsource) [reply eslint-bridge/tests/routing/routing.test.ts:28]: nope, just a intellij proposed fix

File: eslint-bridge/jest.config.js (hunk 1 above)
Comment (victor-diez-sonarsource) [reply eslint-bridge/jest.config.js:18]: this line has no effect it seems: https://github.com/SonarSource/SonarJS/blob/master/eslint-bridge/tests/routing/routing.test.ts#L43

https://github.com/facebook/jest/issues/11607
